lib/natural/distance/jaro-winkler_distance.js

```diff
@@ -26,11 +26,14 @@ THE SOFTWARE.
 // http://en.wikipedia.org/wiki/Jaro%E2%80%93Winkler_distance
 // s1 is the first string to compare
 // s2 is the second string to compare
-function distance(s1, s2) {
+// caseSensitive is a boolean expression determining whether 'a' matches 'A'
+function distance(s1, s2, caseSensitive) {
     if (typeof(s1) != "string" || typeof(s2) != "string") return 0;
     if (s1.length == 0 || s2.length == 0) 
         return 0;
-    s1 = s1.toLowerCase(), s2 = s2.toLowerCase();
+    if (!(caseSensitive))

```

Comment:
Then this would become:

       if(ignoreCase)

Reply:
The distance function <code>distance(s1, s2)</code> is not the right spot to change the case of the strings. The strings are also analysed in the main function <code>JaroWinklerDistance()</code>, and changing case here will not be seen by the main function.

It has to be done right at the beginning of the main function:
```javascript
function JaroWinklerDistance(s1, s2, dj, ignoreCase) {
    if (s1 === s2) {
        return 1;
    } else {
        if (ignoreCase) {
          s1 = s1.toLowerCase();
          s2 = s2.toLowerCase();
        }
```

Hugo